include/camp/resource/cuda.hpp

```diff
@@ -154,31 +176,63 @@ namespace resources
 
       // Memory
       template <typename T>
-      T *allocate(size_t size)
+      T *allocate(size_t size, MemoryAccess ma = MemoryAccess::Device)
       {
         T *ret = nullptr;
         if (size > 0) {
           auto d{device_guard(device)};
-          campCudaErrchk(cudaMallocManaged(&ret, sizeof(T) * size));
+          switch (ma) {
+            case MemoryAccess::Unknown:
+            case MemoryAccess::Device:
+              campCudaErrchk(cudaMalloc(&ret, sizeof(T) * size));
+              break;
+            case MemoryAccess::Pinned:
+              // TODO: do a test here for whether managed is *actually* shared
+              // so we can use the better performing memory
+              campCudaErrchk(cudaMallocHost(&ret, sizeof(T) * size));
+              break;
+            case MemoryAccess::Managed:
+              campCudaErrchk(cudaMallocManaged(&ret, sizeof(T) * size));
+              break;
+          }
         }
         return ret;
       }
-      void *calloc(size_t size)
+      void *calloc(size_t size, MemoryAccess ma = MemoryAccess::Device)
       {
         void *p = allocate<char>(size);
```

Comment:
```suggestion
        void *p = allocate<char>(size, ma);
```

Reply:
This needs to forward/pass the `MemoryAccess` type to the actual `allocate()` call.

Reply:
Thanks @rchen20, can't believe I missed that. 🤦